src/cholesky.jl

```diff
@@ -0,0 +1,65 @@
+import Base: getproperty
+import LinearAlgebra: cholesky, Cholesky, char_uplo, UpperTriangular, LowerTriangular, \,
+    istril, istriu
+
+const KroneckerCholesky{T} = Cholesky{T, <:AbstractKroneckerProduct{T}} where {T}
+
+function cholesky(A::AbstractKroneckerProduct; check=true)
+    P, Q = getmatrices(A)
+    chol_P, chol_Q = cholesky(P; check=true), cholesky(Q; check=true)
+    return Cholesky(chol_P.factors ⊗ chol_Q.factors, 'U', 0)
+end
+
+function Cholesky(factors::KroneckerProduct{T}, uplo::AbstractChar, info::Integer) where {T}
+    return Cholesky{T, typeof(factors)}(factors, uplo, info)
+end
+
+function UpperTriangular(C::AbstractKroneckerProduct)
+    A, B = getmatrices(C)
+    return UpperTriangular(A) ⊗ UpperTriangular(B)
```

Comment:
`UpperTriangular` is the constructor of the `UpperTriangular` type. So I guess this should return an object of that type, and you may wish to consider returning
```julia
UpperTriangular(UpperTriangular(A) ⊗ UpperTriangular(B))
```
? Otherwise, subsequent code will not benefit from upper-triangularizing an `AbstractKroneckerProduct`, it will be "invisible" from the type point of view.